tests/test_code_generator.py

```diff
@@ -189,9 +189,18 @@ def test_templated():
     assert templated_o.xi[0].get() == 11
     assert templated_o.xi[1].get() == 12
 
-    # Test free functions
+    # Test (wrap-attached) free functions = old way to wrap static functions (can only be called with class)
+    if int(str(cython_version).split(".")[0]) < 3:
+        assert templated.computeEight() == 8
+        assert templated_o.computeEight() == 8
+    assert twrapped.Templated.computeEight() == 8
+    assert twrapped.Templated_other.computeEight() == 8
+
+    # Test static functions (can be called with or without object)
     assert templated.computeSeven() == 7
     assert templated_o.computeSeven() == 7
+    assert twrapped.Templated.computeSeven() == 7
```

Comment:
ok this is really cool but do we need a check for the cython version here as well?.

Reply:
No, apparently this would have worked for a long time already. Also with Cython 0. See the test matrix.